sql-cli/sql_cli/__main__.py

```diff
@@ -38,5 +42,50 @@ def generate(
     rprint("The DAG file", dag_file.resolve(), "has been successfully generated. 🎉")
 
 
+@app.command()
+def run(
+    dag_id: str = typer.Argument("dag_id", help="ID of the DAG you want to run", show_default=False),
+    subdir: str = typer.Argument(
+        "subdir",
+        help="the subdirectory or filepath for the DAG. "
+        "The more precise this is the faster the local runner will run as we won't need to parse as many DAGs",
+        show_default=False,
+    ),
+    conn_file_path: Optional[str] = typer.Argument(
+        help="path to connections yaml or json file", default=None
+    ),
+    variable_file_path: Optional[str] = typer.Argument(
+        help="path to variables yaml or json file", default=None
+    ),
+) -> None:
+    """
+    Run a DAG locally. This task assumes that there is a local airflow DB (can be a SQLite file), that has been
```

Comment:
I think we should be avoiding the word DAG where possible in the SQL CLI. If we could focus on workflow, it would be great.

It's also important that the run command is compatible to receive the name of a SQL directory, within `workflows`, and both generates & runs it

Reply:
🤔 but all DAGs are workflows - not all workflows are DAGs - and we only allow workflows which are DAGs to run. So I think DAG is more precisely what we run.

I think we should have in general discussion about terms/names we use in sql-cli. I also thought about renaming some classes and functions to flow/workflow instead of dag and then only the dag_generator is allowed to be called **dag**_generator.

In my opinion this is something we should clean up / make clear before 1.0 definitely.

Reply:
Ah sorry that is not what I meant. It was only about naming.

Reply:
We can only run DAGs right now, because Airflow only runs DAGs. A cyclic dependency for example is not allowed.